File: ios/Classes/VgsTextViewFactory.swift
//
//  VgsTextViewFactory.swift
//  vgscardinfo
//
//  Created by Nehemie Koffi on 12/11/2021.
//

import Foundation
import Flutter
import UIKit

class VgsTextViewFactory: NSObject, FlutterPlatformViewFactory {
    private var messenger: FlutterBinaryMessenger

    init(messenger: FlutterBinaryMessenger) {
        self.messenger = messenger
        super.init()
    }

    func create(
        withFrame frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?
    ) -> FlutterPlatformView {
        return VgsTextView(
            frame: frame,
            binaryMessenger: messenger,
            viewIdentifier: viewId,
            arguments: args as Any
            )
    }
    public func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
            return FlutterStandardMessageCodec.sharedInstance()
        }
}

File: android/src/main/kotlin/com/djamo/plugin/vgscardinfo/VgsTextViewFactory.kt
package com.djamo.plugin.vgscardinfo
import android.content.Context
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class VgsTextViewFactory(private val messenger: BinaryMessenger? = null) :
        PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    override fun create(context: Context?, id: Int, args: Any?): PlatformView {
        val vgsParams = args as Map<String?, Any?>?
        return VgsTextView(context, messenger, id, vgsParams)
    }
}